### ISSUE TITLE
The append method from useListState overrides items in the array on multiple calls in version 4.2.4

### ISSUE THREAD

**Issue by HreneanAlin** [mantinedev/mantine#1431]
### What package has an issue

@mantine/hooks

### Describe the bug

The append method from useListState overrides items in the array on multiple calls inside `useEffect` in version 4.2.4

here is a `codesandbox` example  reproducing the issue
https://codesandbox.io/s/blissful-ramanujan-h934k8?file=/src/index.js

here is a equivalent using version 4.08 that works as expected 
https://codesandbox.io/s/epic-rumple-vm924n?file=/src/index.js 

### In which browser did the problem occur

Chrome 101

### If possible, please include a link to a codesandbox with the reproduced problem

https://codesandbox.io/s/blissful-ramanujan-h934k8?file=/src/index.js

### Possible fix

it may be a regression caused by a recent update

**Comment by rtivital** [comment]
Thanks for reporting

**Comment by nrayburn-tech** [comment]
Looks like a regression from https://github.com/mantinedev/mantine/pull/1387.  Something to do with the difference between `setState([...state, ...items])` and `setState((prev) => [...prev, ...items])`.

I am working on a PR, but not sure of the correct approach just yet.

Test case for the append method, but each of the methods probably need one.
```ts
  it('does not override previous values with append handler', () => {
    const hook = renderHook(() => useListState(['test-1', 'test-2']));
    act(() => {
      const [, handlers] = hook.result.current;
      handlers.append('test-3');
      handlers.append('test-4');
    });

    const [state] = hook.result.current;
    expect(state).toHaveLength(4);
  });
``

**Comment by rtivital** [comment]
It's an issue with recent changes, I'll just roll them back in next patch since they produce issues

**Comment by rtivital** [comment]
Fixed in 4.2.5
